### ISSUE TITLE
PT021 and “factory as fixture” pattern

### ISSUE THREAD

**Issue by skarzi** [m-burst/flake8-pytest-style#46]
Hi,

first of all thanks for awesome project! :+1: 

When updating to `1.2.0` I have spotted 1 issue, mainly, sometimes it's convenient or necessary to use ["factory as a fixture" pattern](https://docs.pytest.org/en/stable/fixture.html#factories-as-fixtures).
In such cases `request.addfinalizer` is often the cleanest solution, because it allows to skip some kind of local registration needed when using `yield` statement. 
To give some example, let's compare this snippet, where `request.addfinalizer` is used:

```python
@pytest.fixture()
def some_factory(request):
    def factory(some_arg: str) -> SomeClass:
        some_class = SomeClass(some_arg)
        request.add_finalizer(some_class.clean_mess)
        return some_class
    return factory
``` 

with the same example, but with `yield`:

```python
@pytest.fixture()
def some_factory(request):
    some_class_registry = []
    def factory(some_arg: str) -> SomeClass:
        some_class = SomeClass(some_arg)
        some_class_registry.append()
        return some_class
    yield factory
    for some_class in reversed(some_class_registry):
        some_class.clean_mess()
```

Currently I have used `noqa` in such cases and it's in majority of cases fine, because it's not so widely used pattern, but I am raising an issue, because maybe you know some better solution for this problem or it's relatively easy to implement (of course I am happy to help!)

**Comment by m-burst** [comment]
Hi @skarzi,

Thanks for your report!
I'll fix the rule soon to allow calls to `request.addfinalizer` in functions nested inside fixtures.

Until then, the solution is either to add `noqa` or to move the nested function outside, e.g.:
```python
import functools
import pytest

def _create_some_class(some_arg: str, request) -> SomeClass:
    some_class = SomeClass(some_arg)
    request.add_finalizer(some_class.clean_mess)
    return some_class

@pytest.fixture()
def some_factory(request):
    return functools.partial(_create_some_class, request=request)
```

**Comment by skarzi** [comment]
Great!
Thanks for quick response
If you will need any help with this feature, just let me know

**Comment by m-burst** [comment]
I've just released v1.2.1 with the fix.
Thanks again for the report and for using the plugin, and please report if you have any further issues :)
